SpockOpaleye/appcore/src/Utils.hs

```diff
@@ -0,0 +1,10 @@
+module Utils where
+
+import UserDefs
+import CryptoDef
+import InternalUtils
+import Data.Text
+import TenantId
+
+fillPassword :: UserIncoming -> BcryptPassword -> UserPoly () () () TenantId Text BcryptPassword (Maybe Text) (Maybe Text) ()
+fillPassword incUser hash = incUser { _userpolyPassword = hash }
```

Comment:
So, ensuring that only `BcryptPassword` can be set is the right thing. But that simply means that the callsite needs to convert `Text -> BcryptPassword` before using the setter. If you're making a convenience function for this -- the convenience function itself should use the setter, which is not happening in this case.